app/common/auth/auth.service.ts

```diff
@@ -0,0 +1,79 @@
+import 'rxjs/Rx';
+
+import {Injectable} from '@angular/core';
+import {CustomHttp} from "../services/CustomHttp";
+import {AuthData} from "./auth.data";
+import {Router} from "@angular/router";
+import {Observable} from "rxjs/Rx";
+
+@Injectable()
+export class AuthService {
+	private _loading: Observable<any>;
+	private _role:string = null;
+
+	constructor(private customHttp:CustomHttp,
+	            private router:Router) {
+		this.restore();
+	}
+
+	get role() {
+		return this._role;
+	}
+	set role(role: string) {
+		this._role = role;
+		localStorage.setItem('auth', this._role);
+	}
+
+	public logIn(loginData:AuthData, errHandler?:(err, data) => void) {
```

Comment:
errHandler better to remove in the future and make logIn return Observable (so than catch errors on it in login component).